content/5_Plat/Bin_Jump.mdx

```diff
@@ -320,10 +317,163 @@ int main() {
 
 <FocusProblem problem={problems.lca[1]} />
 
-### Tutorial
+### Solution (Company Queries II) 
+
+<Resources>
+	<Resource source="CPH" title="18.3 - LCA Method 1">brief description</Resource>
+	<Resource
+		source="Benq"
+		title="LCAjump"
+		url="https://github.com/bqi343/USACO/blob/master/Implementations/content/graphs%20(12)/Trees%20(10)/LCAjump.h"
+	>
+		alternative implementation
+	</Resource>
+</Resources>
+
+### Solution (Company Queries II) 
+
+To find $\textrm{lca}(a, b)$, we can first lift the lower node of $a$ and $b$ to the same depth as the other. Then, we lift both nodes up decrementally. At the end, the parent of either node is the LCA of the two.
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+CodeSnip{Benq Template}
+	
+int N, Q, T=1;
+int depth[200005];
+int up[200005][20];
+vi adj[200005];
+
+void dfs(int v) {
+	FOR(i, 1, 20) {
+		up[v][i] = up[up[v][i-1]][i-1];
+	}
+
+	for(int x : adj[v]) {
+		if(x != up[v][0]) {
+			depth[x] = depth[up[x][0] = v]+1;
+			dfs(x);
+		}
+	}
+}
+
+int jump(int x, int d) {
+	F0R(i, 20) {
+		if ((d >> i) & 1) 
+			x = up[x][i];
+	}
+	return x; 
+}
+
+int LCA(int a, int b) {
+	if(depth[a] < depth[b]) swap(a, b);
+
+	a = jump(a, depth[a] - depth[b]);
+	if(a == b) return a;
+
+	R0F(i, 20) {
+		int aT = up[a][i], bT = up[b][i];
+		if(aT != bT) a = aT, b = bT;
+	}
+
+	return up[a][0];
+}
+	
+int main() {
+	re(N, Q);
+	
+	rep(N-1) {
```

Comment:
why is `rep` used here but `R0F` and `F0R` used in other places?

Reply:
I have a `rep` macro :)

Reply:
well TIL ig :p